jax/experimental/jax2tf/jax2tf.py

```diff
@@ -356,7 +470,7 @@ def post_process_map(self, map_primitive, out_tracers, params):
 
   def get_primitive_impl(self, p):
     try:
-      return tf_impl[p]
+      return Categorizer().wrap(p, tf_impl[p])
```

Comment:
I am not yet sold on the need for the singleton. 
I don't quite see how the categorizer is invoked. Wouldn't it be enough for the test harness to overwrite get_primitive_impl with a categorizer? Then people who read the code don't have to wonder what the categorizer is doing (since it does not matter outside of tests).